pennylane/io/qualtran_io.py

```diff
@@ -0,0 +1,293 @@
+# Copyright 2018-2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This submodule contains the adapter class for Qualtran-PennyLane interoperability.
+"""
+import numpy as np
+
+import pennylane as qml
+from pennylane.operation import DecompositionUndefinedError, MatrixUndefinedError, Operation
+from pennylane.wires import WiresLike
+
+try:
+    import qualtran as qt
+except (ModuleNotFoundError, ImportError) as import_error:
+    pass
+
+
+def get_bloq_registers_info(bloq):
+    """Returns a ``qml.registers`` object associated with all named and unnamed registers and wires
+    in the bloq.
+
+    Args:
+        bloq (Bloq): the bloq to get the registers info of
+
+    Returns:
+        dict: A dictionary that has all the named and un-named registers with default wire
+        ordering.
+
+    Raises:
+        TypeError: bloq must be an instance of :code:`~.Bloq`.
+
+    **Example**
+
+    Given a qualtran bloq:
+
+    >>> from qualtran.bloqs.phase_estimation import RectangularWindowState, TextbookQPE
+    >>> textbook_qpe_small = TextbookQPE(ZPowGate(exponent=2 * 0.234), RectangularWindowState(3))
+    >>> qml.get_bloq_registers_info(textbook_qpe_small)
+    {'qpe_reg': Wires([0, 1, 2]), 'q': Wires([3])}
+    """
+    if not isinstance(bloq, qt.Bloq):
+        raise TypeError(f"bloq must be an instance of {qt.Bloq}.")
+
+    wire_register_dict = {reg.name: reg.bitsize for reg in bloq.signature.rights()}
+
+    return qml.registers(wire_register_dict)
+
+
+def _get_named_registers(registers):
+    """Returns a ``qml.registers`` object associated with the named registers in the bloq"""
+
+    temp_register_dict = {reg.name: reg.total_bits() for reg in registers}
+
+    return qml.registers(temp_register_dict)
+
+
+class FromBloq(Operation):
+    r"""
+    An adapter for using a `Qualtran bloq <https://qualtran.readthedocs.io/en/latest/bloqs/index.html#bloqs-library>`_
+    as a PennyLane :class:`~.Operation`.
+
+    .. note::
+        This class requires the latest version of qualtran. We recommend installing the main
```

Comment:
Be consistent with spelling: Qualtran/qualtran, Bloq/bloq. These have to be consistent everywhere in the PR.

Reply:
I have made it consistent so that we use Qualtran and bloq.